bibcheck/plugins/ostimove.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+##
+## This file is part of Invenio.
+## Copyright (C) 2016 CERN.
+##
+## Invenio is free software; you can redistribute it and/or
+## modify it under the terms of the GNU General Public License as
+## published by the Free Software Foundation; either version 2 of the
+## License, or (at your option) any later version.
+##
+## Invenio is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Invenio; if not, write to the Free Software Foundation, Inc.,
+## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+""" Bibcheck plugin to move info for OSTI from tag
+    8564_w/y to tag 035__a/9
+"""
+
+from invenio.bibrecord import record_add_field
+import re
+
+PROVENANCE = 'OSTI'
+
+OSTI_ID = re.compile(r'^https?://(?:www\.)?osti\.gov/.*?\D(\d+)$')
+
+
+def check_record(record):
+    """ move 8564_u/y to 035__a/9 """
+    for pos, val in record.iterfield('8564_u',
+                                     subfield_filter=('y', PROVENANCE)):
+        if val:
+            ostimatch = OSTI_ID.match(val)
+            if ostimatch:
+                ostimatch_tuple = ostimatch.groups()
+                ostimatch_string = ''.join(ostimatch_tuple)
+                osti_subfields = [('9', 'OSTI'), ('a', ostimatch_string)]
+                record_add_field(record, '035', ' ', ' ',
+                                     subfields=osti_subfields)
+                record.delete_field((pos[0][0:3], pos[1], None))
```

Comment:
I performed a quick and dirty check
```

In [11]: for r in perform_request_search(p="8564_y:osti"):                                                                                                                                               
    fv = get_fieldvalues(r, '8564_u')                                                                                                                                   
    fv = [ore.match(u).group(1) for u in fv if ore.match(u)]                                                                                                            
   ....:     if len(fv) == 1:                                                                                                                                           
   ....:         ws = get_fieldvalues(r, '8564_w')
   ....:         if ws and fv[0] not in ws:                                                                                                                             
   ....:             print r, fv, ws
   ....:             
127467 ['6534741'] ['p7803945']
256548 ['10105486'] ['4252']
417505 ['228486'] ['9921']
463206 ['658361'] ['441/289']
570177 ['789433'] ['p01/PAPERS/RPPH048.PDF']
570194 ['789381'] ['p01/PAPERS/RPPH074.PDF']
848315 ['941166'] ['074/079/Nufact08_079.pdf']
1302269 ['1110727'] ['COOL2013/papers/moam2ha04.pdf']
1302271 ['1113193'] ['PAC2013/papers/thpho05.pdf']
1302275 ['1113191'] ['PAC2013/papers/thpho13.pdf']
1302752 ['1113188'] ['PAC2013/papers/thpba08.pdf']
```
those are all because there are other 8564 tags with $$w, none of them indicate a conflict